mlir/lib/Dialect/QCO/Transforms/Decomposition/Helpers.cpp

```
@@ -0,0 +1,133 @@
+#include <llvm/Support/Casting.h>
+#include <llvm/Support/ErrorHandling.h>
+#include <mlir/IR/Operation.h>
+
+#include <cmath>
+#include <complex>
+#include <cstddef>
+#include <numbers>
+
+namespace mlir::qco::helpers {
+
+decomposition::GateKind getGateKind(UnitaryOpInterface op) {
```

Cppcheck (2.20.0)

[style] The function 'getGateKind' is never used.

(unusedFunction)